packages/playground-examples/copy/ja/TypeScript/Meta-Types/Discriminate Types.ts

```diff
@@ -0,0 +1,66 @@
+// 区別された総称型は
+// コードフロー解析で可能性のあるオブジェクトの種類を
+// 特定の1つのオブジェクトに減らすのに使うものです。
+//
+// このパターンは異なる文字列や数字の定数を持つ
+// よく似たオブジェクトの集合に対してよく機能します。
+// 例: 名前付きイベントのリストや
+// バージョン付けされたオブジェクトの集合
+
+type TimingEvent = { name: "start"; userStarted: boolean } | { name: "closed"; duration: number };
+
+// イベントがこの関数にやってきたとき、
+// 2つの型のどちらの可能性もあります。
+
+const handleEvent = (event: TimingEvent) => {
+  // event.nameに対してswitch文を用いることで、
+  // TypeScriptのコードフロー解析はオブジェクトが
+  // 総称型のうちの片方で表せるということを決定できます。
+
+  switch (event.name) {
+    case "start":
+      // TimingEventのうち、"start"になる型は1つしかないので、
+      // これはuserStartedに
+      // 安全にアクセスできることを意味します。
```

Comment:
[IMO]

プロパティの name についての記載が合ったほうが良いと思いましたがいかがでしょうか？

```suggestion
      // TimingEventのうち、nameが"start"になる型は1つしかないので、
      // これはuserStartedに
      // 安全にアクセスできることを意味します。
```